code/game/objects/structures/safe.dm

```diff
@@ -211,96 +212,80 @@ GLOBAL_LIST_EMPTY(safes)
 	if(total_ticks == 1 || prob(10))
 		to_chat(user, "<span class='italics'>You hear a [pick(sounds)] from [src].</span>")
 
-/obj/structure/safe/Topic(href, href_list)
-	if(..())
-		return TRUE
-
-	var/mob/user = usr
-	if(!user.IsAdvancedToolUser() && !isobserver(user))
-		to_chat(user, "<span class='warning'>You're not able to operate the safe.</span>")
+/obj/structure/safe/proc/start_turning_knob(mob/user, canhear, ticks, left)
```

Comment:
Consider accepting a signed `ticks` value (with positive values meaning turn right, and negative ones - turn left, or vice versa).
Subjectively, it feels a bit cleaner that way, but feel free to leave as is.